src/api/gettrytes.rs

```diff
@@ -91,51 +88,74 @@ impl GetTrytes {
-            .unwrap()
-            .into_rows()
-            .unwrap();
-        let tx = TcpStreamTx::try_from_row(rows.pop().unwrap()).unwrap();
-        let payload_tritbuf = decode_bytes(&tx.payload.into_vec(), PAYLOAD_TRIT_LEN);
-        let address_tritbuf = decode_bytes(&tx.address.into_vec(), ADDRESS_TRIT_LEN);
-        let value = tx.value as i64;
-        let obs_tag_tritbuf = decode_bytes(&tx.obsolete_tag.into_vec(), TAG_TRIT_LEN);
-        let timestamp = tx.timestamp as u64;
-        let current_index = tx.current_index as usize;
-        let last_index = tx.last_index as usize;
-        let bundle_tritbuf = decode_bytes(&tx.bundle.into_vec(), HASH_TRIT_LEN);
-        let tag_tritbuf = decode_bytes(&tx.tag.into_vec(), TAG_TRIT_LEN);
-        let trunk_tritbuf = decode_bytes(&tx.trunk.into_vec(), TAG_TRIT_LEN);
-        let branch_tritbuf = decode_bytes(&tx.branch.into_vec(), TAG_TRIT_LEN);
-        let attachment_ts = tx.attachment_ts as u64;
-        let attachment_lbts = tx.attachment_lbts as u64;
-        let attachment_ubts = tx.attachment_ubts as u64;
-        let nonce_tritbuf = decode_bytes(&tx.nonce.into_vec(), NONCE_TRIT_LEN);
-        let builder = TransactionBuilder::new()
-            .with_payload(Payload::from_inner_unchecked(payload_tritbuf))
-            .with_address(Address::from_inner_unchecked(address_tritbuf))
-            .with_value(Value::from_inner_unchecked(value))
-            .with_obsolete_tag(Tag::from_inner_unchecked(obs_tag_tritbuf))
-            .with_timestamp(Timestamp::from_inner_unchecked(timestamp))
-            .with_index(Index::from_inner_unchecked(current_index))
-            .with_last_index(Index::from_inner_unchecked(last_index))
-            .with_bundle(Hash::from_inner_unchecked(bundle_tritbuf))
-            .with_trunk(Hash::from_inner_unchecked(trunk_tritbuf))
-            .with_branch(Hash::from_inner_unchecked(branch_tritbuf))
-            .with_tag(Tag::from_inner_unchecked(tag_tritbuf))
-            .with_attachment_ts(Timestamp::from_inner_unchecked(attachment_ts))
-            .with_attachment_lbts(Timestamp::from_inner_unchecked(attachment_lbts))
-            .with_attachment_ubts(Timestamp::from_inner_unchecked(attachment_ubts))
-            .with_nonce(Nonce::from_inner_unchecked(nonce_tritbuf));
-        Ok(builder.build().unwrap())
+
+        // TODO: modify the prepare_payload
+        let qf = QueryRef::new(1, &[0; 200]);
+        let (tx, mut rx) = mpsc::unbounded_channel::<BrokerEvent>();
+        let worker = Broker::new(tx, qf);
+        let event = reporter::Event::Request {
+            worker: smallbox!(worker),
+            payload: query_frame,
+        };
+        Ring::send(&self.data_center, self.replica_index, self.token, event);
+
+        while let Some(event) = rx.recv().await {
+            match event {
+                BrokerEvent::Response {
+                    giveload,
+                    query,
```

Comment:
There are lots of unused variables. Are you sure they don't need to prefix with _ ?

Reply:
Corrected, thanks.